drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-kogito-client/src/main/java/org/drools/workbench/screens/scenariosimulation/kogito/client/dmn/KogitoScenarioSimulationBuilder.java

```diff
@@ -0,0 +1,441 @@
+import org.drools.scenariosimulation.api.model.BackgroundDataWithIndex;
+import org.drools.scenariosimulation.api.model.ExpressionIdentifier;
+import org.drools.scenariosimulation.api.model.FactIdentifier;
+import org.drools.scenariosimulation.api.model.FactMapping;
+import org.drools.scenariosimulation.api.model.FactMappingType;
+import org.drools.scenariosimulation.api.model.ScenarioSimulationModel;
+import org.drools.scenariosimulation.api.model.ScenarioWithIndex;
+import org.drools.scenariosimulation.api.model.ScesimDataWithIndex;
+import org.drools.scenariosimulation.api.model.ScesimModelDescriptor;
+import org.drools.scenariosimulation.api.model.Settings;
+import org.drools.scenariosimulation.api.model.Simulation;
+import org.drools.scenariosimulation.api.utils.ScenarioSimulationSharedUtils;
+import org.drools.workbench.scenariosimulation.kogito.marshaller.js.SCESIMMainJs;
+import org.drools.workbench.scenariosimulation.kogito.marshaller.js.callbacks.SCESIMMarshallCallback;
+import org.drools.workbench.scenariosimulation.kogito.marshaller.js.model.JSIScenarioSimulationModelType;
+import org.drools.workbench.scenariosimulation.kogito.marshaller.js.model.SCESIM;
+import org.drools.workbench.scenariosimulation.kogito.marshaller.mapper.JSIName;
+import org.drools.workbench.scenariosimulation.kogito.marshaller.mapper.JsUtils;
+import org.drools.workbench.screens.scenariosimulation.kogito.client.converters.scesim.ApiJSInteropConverter;
+import org.drools.workbench.screens.scenariosimulation.model.typedescriptor.FactModelTree;
+import org.drools.workbench.screens.scenariosimulation.model.typedescriptor.FactModelTuple;
+import org.jboss.errai.common.client.api.RemoteCallback;
+import org.kie.workbench.common.dmn.webapp.kogito.marshaller.js.model.MainJs;
+import org.kie.workbench.common.dmn.webapp.kogito.marshaller.js.model.callbacks.DMN12UnmarshallCallback;
+import org.kie.workbench.common.dmn.webapp.kogito.marshaller.js.model.dmn12.JSITDefinitions;
+import org.uberfire.backend.vfs.Path;
+import org.uberfire.backend.vfs.PathFactory;
+
+import static org.drools.scenariosimulation.api.model.FactMappingType.EXPECT;
+import static org.drools.scenariosimulation.api.model.FactMappingType.GIVEN;
+import static org.drools.scenariosimulation.api.utils.ConstantsHolder.VALUE;
+
+/**
+ * Class used to populate <code>ScenarioSimulationModel</code> inside <code>Callback</code>s
+ * <p>
+ * Mostly copied from <code>org.drools.workbench.screens.scenariosimulation.backend.server.util.SimulationSettingsCreationStrategy</code> and its children.
+ * <p>
+ * Implemented to manage the nested callbacks for <b>DMN</b>
+ */
+public class KogitoScenarioSimulationBuilder {
+
+    @Inject
+    private KogitoDMNService dmnTypeService;
+
+    private static FactMappingType convert(final FactModelTree.Type modelTreeType) {
+        switch (modelTreeType) {
+            case INPUT:
+                return GIVEN;
+            case DECISION:
+                return EXPECT;
```

Comment:
Actually in early phases I think was demonstrated that some 'Decision node' can be in 'Input' columns to allow users something like mocking part of decision model. Is this switch correct?

Reply:
This method is used only during scesim file creation to prepopulate the template. After that user can move a decision as input as you described to mock its value